llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -10959,7 +10978,7 @@ bool ScalarEvolution::SimplifyICmpOperands(CmpPredicate &Pred, const SCEV *&LHS,
   // Recursively simplify until we either hit a recursion limit or nothing
   // changes.
   if (Changed)
-    return SimplifyICmpOperands(Pred, LHS, RHS, Depth + 1);
+    return SimplifyICmpOperands(Pred, LHS, RHS, Depth + 1), true;
 
```

Comment:
Why is this change necessary?

Reply:
SimplifyICmpOperands is supposed to return true when a change is made, but this recursive call (based on `Change==true`) will return false when no further changes are made.

Reply:
Okay, comma expressions are uncommon in LLVM, so kindly `return true` explicitly.